sql/src/main/java/org/opensearch/sql/sql/antlr/AnonymizerListener.java

```diff
@@ -0,0 +1,96 @@
+package org.opensearch.sql.sql.antlr;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.ErrorNode;
+import org.antlr.v4.runtime.tree.ParseTreeListener;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.*;
+
+public class AnonymizerListener implements ParseTreeListener {
+  private String anonymizedQueryString = "";
+  private int previousType = -1;
+
+  @Override
+  public void enterEveryRule(ParserRuleContext ctx) {
+  }
+
+  @Override
+  public void exitEveryRule(ParserRuleContext ctx) {
+  }
+
+  @Override
+  public void visitTerminal(TerminalNode node) {
+    if (!(node.getSymbol().getType() == RR_BRACKET
```

Comment:
For now I have taken out this logic because it is very complicated and has a lot of edge cases that will be missed through it. For now there is a space between every node except for before and after a dot, and before a comma. With this simple formatting we can get outputs that look reasonable an well formatted, and I will keep my eye out for anything else to make the formatting better without overcomplicating it.

Reply:
This complex statement for checking where to add/remove spaces for the anonymized sql statement has been overhauled to be more simple, as well as use the method that you suggested.